applications/RomApplication/tests/test_randomized_singular_value_decomposition_mpi.py

```diff
@@ -0,0 +1,81 @@
+#import python class test
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+
+#import python packages
+try:
+    from mpi4py import MPI
+    import numpy as np
+    import numpy.matlib
+    import KratosMultiphysics.RomApplication.tsqr
+    from KratosMultiphysics.RomApplication.randomized_singular_value_decomposition import RandomizedSingularValueDecomposition
+    numpy_and_mpi4py_available = True
+except:
+    numpy_and_mpi4py_available  = False
+
+def synthetic_matrix(degree, rows = 40,repetitions=4):
+    TestMatrix = np.zeros((rows,degree))
+    x = np.linspace(0,1,rows)
+    for i in range(degree):
+        TestMatrix[:,i] = np.power(x,i)
+    return np.matlib.repmat(TestMatrix, 1, repetitions)
+
+class TestRandomizedSVDMPI(KratosUnittest.TestCase):
+
+    @KratosUnittest.skipUnless(numpy_and_mpi4py_available, "numpy and mpi4py are required for TSQR MPI")
+    def test_radomized_svd(self):
```

Comment:
Unit tests should be deterministic. The random algorithm should be seeded with some arbitrary (but consistent) value at the begining of every test.

Reply:
It is already being seeded ("0") since the mpi algorithm requires it for all cases. In addition, note that it's being compared by assertLess to a given tolerance, this must be satisfied for random theory in all cases. Referring to Brunton's book "There is an extensive literature on random matrix theory, where the above stereotypes (this case) are almost certainly true, meaning that they are true with high probability".

Reply:
Oh I couldn't find the seeding with a quick scan. Good then.